Implement model, model group and connector apis

acha5066 opened the pull request:
### Description
This PR implements the model, model group and connector apis from the ML Commons API so that developers can set up connectors to 3rd party tools like OpenAI as described here: https://github.com/opensearch-project/ml-commons/blob/main/docs/remote_inference_blueprints/openai_connector_embedding_blueprint.md

### Issues Resolved
Closes #169 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

dblock reviewed (commented):
A cursory look at it, looks ok. It needs tests and documentation, please. I would write a working sample as well. Sign commits with `-s` to pass DCO.

I encourage anyone wanting to add APIs to spend time doing https://github.com/opensearch-project/opensearch-php/issues/97 instread of hand-rolling these.

acha5066 commented:
* Added tests for the new namespace methods. 
* Added sample usage to `USER_GUIDE.md.`
* Added documentation to the namespace class header. Linked to relevant API docs. Wasn't sure what documentation you had in mind so please let me know. 

Regarding DCO I did signoff with Andrew Chappell and changed my username like: `git config --global user.name "Andrew Chappell"` so not sure why that isn't reflected. It keeps associating with andrew.chappell :thinking:

dblock reviewed (commented):
I made a bunch of nitpicky comments. @shyim should take a look at the code.

For DCO I think you will have to amend previous commits. I would make a new branch off main and merge the code and re-commit it and force push the branch. LMK if you need help on how to do that!

acha5066 commented:
DCO should be fine now

dblock commented:
@shyim Got some time to CR this?

shyim reviewed (approved):
Looks good
